# Create separate csv files for a few subreddits

This project will look at classifying subreddit threads as either Spacex or not Spacex. We'll build a corpus from several subreddits for use in various ways.  This particular notebook will create the datasets

In [1]:
import pandas as pd
import numpy as np
import lebowski as dude

In [2]:
#Create the dataframe for the spacex 'submission' post type

spacex = dude.create_lexicon('spacex', 5000)

connection nominal
Added in 100 new rows
List contains 100 rows
connection nominal
Added in 100 new rows
List contains 200 rows
connection nominal
Added in 100 new rows
List contains 300 rows
connection nominal
Added in 100 new rows
List contains 400 rows
connection nominal
Added in 100 new rows
List contains 500 rows
connection nominal
Added in 100 new rows
List contains 600 rows
connection nominal
Added in 100 new rows
List contains 700 rows
connection nominal
Added in 100 new rows
List contains 800 rows
connection nominal
Added in 100 new rows
List contains 900 rows
connection nominal
Added in 100 new rows
List contains 1000 rows
connection nominal
Added in 100 new rows
List contains 1100 rows
connection nominal
Added in 100 new rows
List contains 1200 rows
connection nominal
Added in 100 new rows
List contains 1300 rows
connection nominal
Added in 100 new rows
List contains 1400 rows
connection nominal
Added in 100 new rows
List contains 1500 rows
connection nominal
Added in 100 ne

In [3]:
spacex_comments = dude.create_lexicon('spacex', 5000, post_type='comment')

connection nominal
Added in 100 new rows
List contains 100 rows
connection nominal
Added in 100 new rows
List contains 200 rows
connection nominal
Added in 100 new rows
List contains 300 rows
connection nominal
Added in 100 new rows
List contains 400 rows
connection nominal
Added in 100 new rows
List contains 500 rows
connection nominal
Added in 100 new rows
List contains 600 rows
connection nominal
Added in 100 new rows
List contains 700 rows
connection nominal
Added in 100 new rows
List contains 800 rows
connection nominal
Added in 100 new rows
List contains 900 rows
connection nominal
Added in 100 new rows
List contains 1000 rows
connection nominal
Added in 100 new rows
List contains 1100 rows
connection nominal
Added in 100 new rows
List contains 1200 rows
connection nominal
Added in 100 new rows
List contains 1300 rows
connection nominal
Added in 100 new rows
List contains 1400 rows
connection nominal
Added in 100 new rows
List contains 1500 rows
connection nominal
Added in 100 ne

In [4]:
virgingalactic = dude.create_lexicon('virgingalactic', 5000)

connection nominal
Added in 100 new rows
List contains 100 rows
connection nominal
Added in 100 new rows
List contains 200 rows
connection nominal
An error occurred, possibly there are not enough posts to scrape, 
        but maybe something completely different
Creating dataframe
Printing dataframe shape:
(200, 73)


In [32]:
virgingalactic = dude.create_lexicon('virgingalactic', 5000)

connection nominal
Added in 100 new rows
List contains 100 rows
connection nominal
Added in 100 new rows
List contains 200 rows
connection nominal
Added in 100 new rows
List contains 300 rows
connection nominal
Added in 100 new rows
List contains 400 rows
connection nominal
Added in 100 new rows
List contains 500 rows
connection nominal
Added in 24 new rows
List contains 524 rows
connection nominal
Added in 0 new rows
List contains 524 rows
An error occurred, possibly there are not enough posts to scrape, 
        but maybe something completely different
Creating dataframe
Printing dataframe shape:
(524, 89)


In [31]:
virgingalactic_comments = dude.create_lexicon('virgingalactic', 5000, post_type='comment')

connection nominal
Added in 100 new rows
List contains 100 rows
connection nominal
Added in 100 new rows
List contains 200 rows
connection nominal
Added in 100 new rows
List contains 300 rows
connection nominal
Added in 100 new rows
List contains 400 rows
connection nominal
Added in 100 new rows
List contains 500 rows
connection nominal
Added in 100 new rows
List contains 600 rows
connection nominal
Added in 100 new rows
List contains 700 rows
connection nominal
Added in 100 new rows
List contains 800 rows
connection nominal
Added in 100 new rows
List contains 900 rows
connection nominal
Added in 100 new rows
List contains 1000 rows
connection nominal
Added in 100 new rows
List contains 1100 rows
connection nominal
Added in 100 new rows
List contains 1200 rows
connection nominal
Added in 100 new rows
List contains 1300 rows
connection nominal
Added in 100 new rows
List contains 1400 rows
connection nominal
Added in 100 new rows
List contains 1500 rows
connection nominal
Added in 100 ne

Tried both VirginGalactic comment and submission subreddits, and each have only 524 posts. It is possible that there won't be enough data here to use.

In [33]:
boeing = dude.create_lexicon('boeing', 5000)

connection nominal
Added in 100 new rows
List contains 100 rows
connection nominal
Added in 100 new rows
List contains 200 rows
connection nominal
Added in 100 new rows
List contains 300 rows
connection nominal
Added in 100 new rows
List contains 400 rows
connection nominal
Added in 100 new rows
List contains 500 rows
connection nominal
Added in 100 new rows
List contains 600 rows
connection nominal
Added in 100 new rows
List contains 700 rows
connection nominal
Added in 100 new rows
List contains 800 rows
connection nominal
Added in 100 new rows
List contains 900 rows
connection nominal
Added in 100 new rows
List contains 1000 rows
connection nominal
Added in 100 new rows
List contains 1100 rows
connection nominal
Added in 100 new rows
List contains 1200 rows
connection nominal
Added in 100 new rows
List contains 1300 rows
connection nominal
Added in 100 new rows
List contains 1400 rows
connection nominal
Added in 100 new rows
List contains 1500 rows
connection nominal
Added in 100 ne

In [34]:
boeing_comments = dude.create_lexicon('boeing', 5000, post_type='comment')

connection nominal
Added in 100 new rows
List contains 100 rows
connection nominal
Added in 100 new rows
List contains 200 rows
connection nominal
Added in 100 new rows
List contains 300 rows
connection nominal
Added in 100 new rows
List contains 400 rows
connection nominal
Added in 100 new rows
List contains 500 rows
connection nominal
Added in 100 new rows
List contains 600 rows
connection nominal
Added in 100 new rows
List contains 700 rows
connection nominal
Added in 100 new rows
List contains 800 rows
connection nominal
Added in 100 new rows
List contains 900 rows
connection nominal
Added in 100 new rows
List contains 1000 rows
connection nominal
Added in 100 new rows
List contains 1100 rows
connection nominal
Added in 100 new rows
List contains 1200 rows
connection nominal
Added in 100 new rows
List contains 1300 rows
connection nominal
Added in 100 new rows
List contains 1400 rows
connection nominal
Added in 100 new rows
List contains 1500 rows
connection nominal
Added in 100 ne

In [35]:
blueorigin = dude.create_lexicon('blueorigin', 5000)

connection nominal
Added in 100 new rows
List contains 100 rows
connection nominal
Added in 100 new rows
List contains 200 rows
connection nominal
Added in 100 new rows
List contains 300 rows
connection nominal
Added in 100 new rows
List contains 400 rows
connection nominal
Added in 100 new rows
List contains 500 rows
connection nominal
Added in 100 new rows
List contains 600 rows
connection nominal
Added in 100 new rows
List contains 700 rows
connection nominal
Added in 100 new rows
List contains 800 rows
connection nominal
Added in 98 new rows
List contains 898 rows
connection nominal
Added in 100 new rows
List contains 998 rows
connection nominal
Added in 100 new rows
List contains 1098 rows
connection nominal
Added in 100 new rows
List contains 1198 rows
connection nominal
Added in 100 new rows
List contains 1298 rows
connection nominal
Added in 100 new rows
List contains 1398 rows
connection nominal
Added in 100 new rows
List contains 1498 rows
connection nominal
Added in 100 new 

In [36]:
blueorigin_comments = dude.create_lexicon('blueorigin', 5000, post_type='comment')

connection nominal
Added in 100 new rows
List contains 100 rows
connection nominal
Added in 100 new rows
List contains 200 rows
connection nominal
Added in 100 new rows
List contains 300 rows
connection nominal
Added in 100 new rows
List contains 400 rows
connection nominal
Added in 100 new rows
List contains 500 rows
connection nominal
Added in 100 new rows
List contains 600 rows
connection nominal
Added in 100 new rows
List contains 700 rows
connection nominal
Added in 100 new rows
List contains 800 rows
connection nominal
Added in 100 new rows
List contains 900 rows
connection nominal
Added in 100 new rows
List contains 1000 rows
connection nominal
Added in 100 new rows
List contains 1100 rows
connection nominal
Added in 100 new rows
List contains 1200 rows
connection nominal
Added in 100 new rows
List contains 1300 rows
connection nominal
Added in 100 new rows
List contains 1400 rows
connection nominal
Added in 100 new rows
List contains 1500 rows
connection nominal
Added in 100 ne

Use the following function to erradicate null values from 'body' column once created.

In [37]:
def strip_and_rep(word):
    if len(str(word).strip().replace(" ", "")) < 1:
        return 'replace_me'
    else:
        return word

In [42]:
spacex['selftext'] = spacex['selftext'].map(strip_and_rep)

In [43]:
boeing['selftext'] = boeing['selftext'].map(strip_and_rep)

In [44]:
blueorigin['selftext'] = blueorigin['selftext'].map(strip_and_rep)

In [48]:
spacex.rename(columns={'selftext': 'body'}, inplace=True)

In [46]:
boeing.rename(columns={'selftext': 'body'}, inplace=True)

In [47]:
blueorigin.rename(columns={'selftext': 'body'}, inplace=True)

# Save the features that we'll use, then concat the df's

In [49]:
spacex_comments.body

0                                               [deleted]
1       Tank didn't frost so I think that means they d...
2       thank you! awesome resource for those who like...
3       If they got the data they wanted and everythin...
4                                 so, elon musk will fly?
                              ...                        
2995                                            [removed]
2996    Yeah I definitely didn't think they were going...
2997    Medical emergency would probably mean an insta...
2998    Hawthorne, California. There is a giant buildi...
2999    Acronyms, initialisms, abbreviations, contract...
Name: body, Length: 3000, dtype: object

In [50]:
features = ['subreddit', 'body', 'permalink']
spacex_comments[features]

subreddit                                               body  \
0       spacex                                          [deleted]   
1       spacex  Tank didn't frost so I think that means they d...   
2       spacex  thank you! awesome resource for those who like...   
3       spacex  If they got the data they wanted and everythin...   
4       spacex                            so, elon musk will fly?   
...        ...                                                ...   
2995    spacex                                          [removed]   
2996    spacex  Yeah I definitely didn't think they were going...   
2997    spacex  Medical emergency would probably mean an insta...   
2998    spacex  Hawthorne, California. There is a giant buildi...   
2999    spacex  Acronyms, initialisms, abbreviations, contract...   

                                              permalink  
0     /r/spacex/comments/hf368o/starship_development...  
1     /r/spacex/comments/hf368o/starship_development...  
2     /r/spacex/comments/hf368o/starship_development...  
3     /r/spacex/comments/hf368o/starship_development...  
4     /r/spacex/comments/hvbja9/elon_musk_will_attem...  
...                                                 ...  
2995  /r/spacex/comments/hsbm9f/falcon_9_rocket_arri...  
2996  /r/spacex/comments/hf368o/starship_development...  
2997  /r/spacex/comments/hsbm9f/falcon_9_rocket_arri...  
2998  /r/spacex/comments/hsbm9f/falcon_9_rocket_arri...  
2999  /r/spacex/comments/hsbm9f/falcon_9_rocket_arri...  

[3000 rows x 3 columns]

In [51]:
boeing_comments[features]

subreddit                                               body  \
0       boeing                  The stupidity in this one comment   
1       boeing                                             Thanks   
2       boeing  Hey all,\n\nI was recently hired for a softwar...   
3       boeing  Your [submission](https://www.reddit.com/r/boe...   
4       boeing  It definitely won't go up, but probably will s...   
...        ...                                                ...   
2395    boeing  There’s people who have been there for 10 year...   
2396    boeing  My manager basically told me I have free reign...   
2397    boeing  Which is good news for me. I’ve only been a bl...   
2398    boeing  I could handle a year or less. But it sounds l...   
2399    boeing  Holy shit, I had this same exact experience. \...   

                                              permalink  
0     /r/boeing/comments/hvucq5/boeing_aircraft_from...  
1     /r/boeing/comments/hvucq5/boeing_aircraft_from...  
2     /r/boeing/comments/hhz2g3/employeeemployment_q...  
3     /r/boeing/comments/hw6zqc/new_grad_hireorienta...  
4     /r/boeing/comments/ha69oj/speea_engineer_trans...  
...                                                 ...  
2395  /r/boeing/comments/goqsco/out_of_sequence_rete...  
2396  /r/boeing/comments/goqsco/out_of_sequence_rete...  
2397  /r/boeing/comments/goqsco/out_of_sequence_rete...  
2398  /r/boeing/comments/goqsco/out_of_sequence_rete...  
2399  /r/boeing/comments/goqsco/out_of_sequence_rete...  

[2400 rows x 3 columns]

In [52]:
blueorigin_comments[features]

subreddit                                               body  \
0     BlueOrigin  I don't know why they would want to waste prop...   
1     BlueOrigin           Haha what if we stole one of his houses?   
2     BlueOrigin  He made a comment about New Glenn being late w...   
3     BlueOrigin                                Bezos such a tease.   
4     BlueOrigin  Engineering interviews typically assess your c...   
...          ...                                                ...   
1995  BlueOrigin  Is this the reduction to 2, or the reduction t...   
1996  BlueOrigin  Given how much more political weight this has,...   
1997  BlueOrigin  Whenever you have this level of change, there ...   
1998  BlueOrigin  Not 100% sure but I think the companies couldn...   
1999  BlueOrigin  I don’t think this award will be for just a mo...   

                                              permalink  
0     /r/BlueOrigin/comments/hoc1in/is_there_a_blue_...  
1     /r/BlueOrigin/comments/hus5ng/just_posted_a_ne...  
2     /r/BlueOrigin/comments/gg9748/official_questio...  
3     /r/BlueOrigin/comments/hsm4um/lc36_progress_li...  
4     /r/BlueOrigin/comments/hu1qh8/upcoming_technic...  
...                                                 ...  
1995  /r/BlueOrigin/comments/ga3owd/nasa_to_announce...  
1996  /r/BlueOrigin/comments/ga3owd/nasa_to_announce...  
1997  /r/BlueOrigin/comments/ga3owd/nasa_to_announce...  
1998  /r/BlueOrigin/comments/ga3owd/nasa_to_announce...  
1999  /r/BlueOrigin/comments/ga3owd/nasa_to_announce...  

[2000 rows x 3 columns]

Concatenate each company's posts into a dataframe

In [53]:
spacex_df = pd.concat([spacex[features], spacex_comments[features]])
spacex_df.shape

(8000, 3)

In [54]:
boeing_df = pd.concat([boeing[features], boeing_comments[features]])
boeing_df.shape

(5848, 3)

In [55]:
blueorigin_df = pd.concat([blueorigin[features], blueorigin[features]])
blueorigin_df.shape

(3288, 3)

## Now send each of these dataframes to a csv file in the ./data/ folder of this directory

In [56]:
spacex_df.to_csv('./data/spacex.csv', index=False)

In [57]:
boeing_df.to_csv('./data/boeing.csv', index=False)

In [58]:
blueorigin_df.to_csv('./data/blueorigin.csv', index=False)

## Random test sample: Can we take a slice of each list and put it into a test set?  That would then give the model a jumble that included data from Blue Origin's subreddit that was not in the training data at all. It would still be a binary classifier, because I'm predicting 'Yes' for Spacex and 'No' for Boeing and 'No for Blue Origin.